frontend/src/components/Header/Header.js

```diff
@@ -0,0 +1,62 @@
+import React, { useContext, useEffect } from "react"
+import { Link, useNavigate } from "react-router-dom"
+import headerstyle from "./Header.module.css"
+import { UserContext } from "../../App"
+
+const Header = () => {
+  const navigate = useNavigate()
+  const { state, dispatch } = useContext(UserContext)
+
+  useEffect(() => {
```

Comment:
Why useEffect is used?

Reply:
useEffect is used to tell what the Header needs to do after rendering.

Reply:
Make sense. Can you try to comment down the useEffect and check whether the website behaves as intended?

Reply:
it behaves as intended right now, but if we call an api which returns data before the next page is loaded, then useEffect comes into use, in login, register there are no changes, but when displaying of passes will come into part on the profile page, then it will be needed, so that the data is fetched before the profile page is displayed.

Reply:
ohhh okay.